File: skyspacefourtyfour_ui/src/services/StaffService.js
import http from "../http-common"; 

class StaffService {
  getAllStaffs(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/staff/staffs`, searchDTO);
  }

  get(staffId) {
    return this.getRequest(`/staff/${staffId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/staff?field=${matchData}`, null);
  }

  addStaff(data) {
    return http.post("/staff/addStaff", data);
  }

  update(data) {
  	return http.post("/staff/updateStaff", data);
  }
  
  uploadImage(data,staffId) {
  	return http.postForm("/staff/uploadImage/"+staffId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new StaffService();

File: src/main/java/com/skyspacefourtyfour/service/StaffService.java
package com.skyspacefourtyfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefourtyfour.domain.Staff;
import com.skyspacefourtyfour.dto.StaffDTO;
import com.skyspacefourtyfour.dto.StaffSearchDTO;
import com.skyspacefourtyfour.dto.StaffPageDTO;
import com.skyspacefourtyfour.dto.StaffConvertCriteriaDTO;
import com.skyspacefourtyfour.service.GenericService;
import com.skyspacefourtyfour.dto.common.RequestDTO;
import com.skyspacefourtyfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface StaffService extends GenericService<Staff, Integer> {

	List<Staff> findAll();

	ResultDTO addStaff(StaffDTO staffDTO, RequestDTO requestDTO);

	ResultDTO updateStaff(StaffDTO staffDTO, RequestDTO requestDTO);

    Page<Staff> getAllStaffs(Pageable pageable);

    Page<Staff> getAllStaffs(Specification<Staff> spec, Pageable pageable);

	ResponseEntity<StaffPageDTO> getStaffs(StaffSearchDTO staffSearchDTO);
	
	List<StaffDTO> convertStaffsToStaffDTOs(List<Staff> staffs, StaffConvertCriteriaDTO convertCriteria);

	StaffDTO getStaffDTOById(Integer staffId);







}





